From dtor_core@ameritech.net  Tue May 30 15:33:17 2006
From: Dmitry Torokhov <dtor_core@ameritech.net>
Date: Sat, 29 Apr 2006 05:12:44 +0000 (-0400)
Subject: Input: psmouse - fix new device detection logic

Input: psmouse - fix new device detection logic

Reported to fix http://bugs.gentoo.org/130846

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Cc: Daniel Drake <dsd@gentoo.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 drivers/input/mouse/psmouse-base.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- linux-2.6.16.18.orig/drivers/input/mouse/psmouse-base.c
+++ linux-2.6.16.18/drivers/input/mouse/psmouse-base.c
@@ -300,8 +300,10 @@ static irqreturn_t psmouse_interrupt(str
  * Check if this is a new device announcement (0xAA 0x00)
  */
 	if (unlikely(psmouse->packet[0] == PSMOUSE_RET_BAT && psmouse->pktcnt <= 2)) {
-		if (psmouse->pktcnt == 1)
+		if (psmouse->pktcnt == 1) {
+			psmouse->last = jiffies;
 			goto out;
+		}
 
 		if (psmouse->packet[1] == PSMOUSE_RET_ID) {
 			__psmouse_set_state(psmouse, PSMOUSE_IGNORE);
